### PR TITLE
Fetch input data

### PR DESCRIPTION
**Pull request by a-solovyev12**
- create component for downloading data in a standardised manner as first step of workflow 
- adapt the channels and IO of subsequent processes to work with downloaded data 
- fix some minor bugs

### CODE REVIEW

File: bin/get_input_data.R [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+suppressPackageStartupMessages(require(optparse))
+suppressPackageStartupMessages(require(workflowscriptscommon))
+suppressPackageStartupMessages(require(R.utils))
+
+option_list = list(
+    make_option(
+        c("-a", "--reference-mat-url"),
+        action = "store",
+        default = NA,
+        type = 'character',
```
Comment (pinin4fjords) [inline bin/get_input_data.R:19]: Why no help text?

Comment (a-solovyev12) [reply bin/get_input_data.R:19]: forgot to add these fields, will fix now

File: bin/get_input_data.R [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# standard file names
+names = c(rep(c("matrix.mtx", "barcodes.tsv",  "genes.tsv"), 2), "ref_metadata.txt", "ref_marker_genes.txt")
+# set up directories
+ref_dir = "reference_10X_dir/"
+query_dir = "query_10X_dir/"
+dir.create(ref_dir)
+dir.create(query_dir)
+
+for(idx in 1:length(names)){
+    if(idx <=3) d = paste0(ref_dir, names[idx], sep = "")
```
Comment (pinin4fjords) [inline bin/get_input_data.R]: Could you add code comments describing what this logic is for please?

Comment (pinin4fjords) [reply bin/get_input_data.R]: Actually, this logic depending on indices is rather convoluted- could you simplify? Maybe what is required in fact is a separate script that creates the 10X directory structure for a single dataset, with parameters? Then you could have separate workflow processes which retrieve the index and query sets. Plus you'd have a re-usable component. 

@pcm32 hacked something together for Galaxy: https://github.com/ebi-gene-expression-group/container-galaxy-sc-tertiary/blob/develop/tools/tertiary-analysis/data-scxa/retrieve-scxa.xml. Maybe we could have an analagous but tidier thing for the CLI layer.

**Pull request closed without merging**